app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -93,6 +95,13 @@ protected void onCreate(final Bundle savedInstanceState) {
         } catch (IOException e) {
             LOGGER.info("Bluetooth handler not started, most likely Bluetooth is not enabled");
             this.enableBluetooth(false);
+        } catch (NullPointerException e) {
+            LOGGER.error("Could not start the Bluetooth service", e);
+            new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                    .setTitleText(this.getString(R.string.something_went_wrong))
+                    .setContentText(this.getString(R.string.something_went_wrong_QR_popup_explanation))
```

Comment:
Is this an accurate message for the problem? Does it only occur when the contact could actually not be exchanged via QR?